Python packages set: major upgrades

FRidh opened the pull request:
<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!
List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

###### Motivation for this change

https://hydra.nixos.org/jobset/nixpkgs/python-unstable


###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

## TODO

- [ ] check nixops
- [ ] keep old arrow around? https://github.com/NixOS/nixpkgs/issues/117227

dotlambda committed:
python3.pkgs.keyring: 23.0.0 -> 23.0.1

fabaff committed:
python3Packages.importlib-metadata: 1.7.0 -> 3.7.2

FRidh committed:
python3Packages.pynisher: 0.6.3 -> 0.6.4

FRidh committed:
python3Packages.pynvim: 0.4.2 -> 0.4.3

FRidh committed:
python3Packages.pynamodb: 4.3.3 -> 5.0.3

FRidh committed:
python3Packages.pyopencl: 2020.3.1 -> 2021.1.4

FRidh committed:
python3Packages.pyobjc: 7.0.1 -> 7.1

FRidh committed:
python3Packages.pytest-order: 0.9.5 -> 0.10.0

FRidh committed:
python3Packages.pyroma: 2.6.1 -> 3.1

FRidh committed:
python3Packages.pyro-ppl: 1.5.2 -> 1.6.0

FRidh committed:
python3Packages.pyspark: 3.0.2 -> 3.1.1

FRidh committed:
python3Packages.pyperclip: 1.8.1 -> 1.8.2

FRidh committed:
python3Packages.pyperf: 2.0.0 -> 2.1.0

FRidh committed:
python3Packages.pytest-mypy: 0.8.0 -> 0.8.1

FRidh committed:
python3Packages.pytest-flask: 1.1.0 -> 1.2.0

FRidh committed:
python3Packages.Pyro5: 5.11 -> 5.12

FRidh committed:
python3Packages.pyramid: 1.10.5 -> 2.0

FRidh committed:
python3Packages.pytest-doctestplus: 0.8.0 -> 0.9.0

FRidh committed:
python3Packages.pytest-isort: 1.2.0 -> 1.3.0

FRidh committed:
python3Packages.pytest-cov: 2.10.1 -> 2.11.1

FRidh committed:
python3Packages.pytest-xprocess: 0.17.0 -> 0.17.1

FRidh committed:
python3Packages.python-ptrace: 0.9.7 -> 0.9.8

FRidh committed:
python3Packages.python-gitlab: 2.5.0 -> 2.6.0

FRidh committed:
python3Packages.python-stdnum: 1.14 -> 1.16

FRidh committed:
python3Packages.pytest-subtesthack: 0.1.1 -> 0.1.2

FRidh committed:
python3Packages.python-gnupg: 0.4.6 -> 0.4.7

FRidh committed:
python3Packages.python-utils: 2.4.0 -> 2.5.6

FRidh committed:
python3Packages.pytest-runner: 5.2 -> 5.3.0

FRidh committed:
python3Packages.python-rapidjson: 0.9.4 -> 1.0

FRidh committed:
python3Packages.python-daemon: 2.2.4 -> 2.3.0

FRidh committed:
python3Packages.python-periphery: 2.1.1 -> 2.3.0

FRidh committed:
python3Packages.rebulk: 2.0.1 -> 3.0.1

FRidh committed:
python3Packages.readthedocs-sphinx-ext: 2.1.3 -> 2.1.4

FRidh committed:
python3Packages.pytools: 2020.4.4 -> 2021.2.1

FRidh committed:
python3Packages.regex: 2020.11.13 -> 2021.3.17

FRidh committed:
python3Packages.pyviz_comms: 0.7.6 -> 2.0.1

FRidh committed:
python3Packages.pyvex: 9.0.5903 -> 9.0.6588

FRidh committed:
python3Packages.py-vapid: 1.7.1 -> 1.8.0

FRidh committed:
python3Packages.radio_beam: 0.3.2 -> 0.3.3

FRidh committed:
python3Packages.pyzmq: 20.0.0 -> 22.0.3

FRidh committed:
python3Packages.reportlab: 3.5.59 -> 3.5.66

FRidh committed:
python3Packages.qtconsole: 5.0.2 -> 5.0.3

FRidh committed:
python3Packages.responses: 0.12.1 -> 0.13.1

FRidh committed:
python3Packages.pywebpush: 1.11.0 -> 1.13.0

FRidh committed:
python3Packages.robotframework: 3.2.2 -> 4.0

FRidh committed:
python3Packages.sentinel: 0.1.2 -> 0.3.0

FRidh committed:
python3Packages.robotframework-sshlibrary: 3.5.1 -> 3.6.0

FRidh committed:
python3Packages.scikit-image: 0.17.2 -> 0.18.1

FRidh committed:
python3Packages.rsa: 4.6 -> 4.7.2

FRidh committed:
python3Packages.sacn: 1.4.6 -> 1.6.2

FRidh committed:
python3Packages.scikit-fmm: 2019.1.30 -> 2021.2.2

FRidh committed:
python3Packages.sanic: 20.9.1 -> 21.3.2

FRidh committed:
python3Packages.rpy2: 3.4.2 -> 3.4.3

FRidh committed:
python3Packages.s3transfer: 0.3.4 -> 0.3.6

FRidh committed:
python3Packages.ruamel.yaml: 0.16.12 -> 0.16.13

FRidh committed:
python3Packages.schedule: 0.6.0 -> 1.0.0

FRidh committed:
python3Packages.robotframework-requests: 0.7.2 -> 0.8.1

FRidh committed:
python3Packages.setuptools_scm: 4.1.2 -> 6.0.1

FRidh committed:
python3Packages.singledispatch: 3.4.0.3 -> 3.6.1

FRidh committed:
python3Packages.smmap: 3.0.5 -> 4.0.0

FRidh committed:
python3Packages.soupsieve: 2.0.1 -> 2.2.1

FRidh committed:
python3Packages.sphinx-autobuild: 2020.9.1 -> 2021.3.14

FRidh committed:
python3Packages.smart_open: 4.1.2 -> 4.2.0

FRidh committed:
python3Packages.spacy: 2.3.5 -> 3.0.5

FRidh committed:
python3Packages.snowballstemmer: 2.0.0 -> 2.1.0

FRidh committed:
python3Packages.sparse: 0.11.2 -> 0.12.0

FRidh committed:
python3Packages.snowflake-connector-python: 2.3.10 -> 2.4.1

FRidh committed:
python3Packages.skein: 0.8.0 -> 0.8.1

FRidh committed:
python3Packages.skorch: 0.9.0 -> 0.10.0

FRidh committed:
python3Packages.stripe: 2.55.2 -> 2.56.0

FRidh committed:
python3Packages.spyder: 4.2.1 -> 4.2.4

FRidh committed:
python3Packages.structlog: 20.1.0 -> 21.1.0

FRidh committed:
python3Packages.spyder-kernels: 1.10.1 -> 1.10.2

FRidh committed:
python3Packages.sunpy: 2.0.7 -> 2.1.2

FRidh committed:
python3Packages.sqlalchemy-citext: 1.7.0 -> 1.8.0

FRidh committed:
python3Packages.sphinx_rtd_theme: 0.4.3 -> 0.5.1

FRidh committed:
python3Packages.sphinxcontrib-bibtex: 2.1.4 -> 2.2.0

FRidh committed:
python3Packages.stevedore: 3.2.2 -> 3.3.0

FRidh committed:
python3Packages.SQLObject: 3.8.1 -> 3.9.1

FRidh committed:
python3Packages.sphinx-autoapi: 1.5.1 -> 1.7.0

FRidh committed:
python3Packages.sphinxcontrib-plantuml: 0.19 -> 0.20

FRidh committed:
python3Packages.testfixtures: 6.15.0 -> 6.17.1

FRidh committed:
python3Packages.svg.path: 4.0.2 -> 4.1

FRidh committed:
python3Packages.timezonefinder: 4.5.0 -> 5.2.0

FRidh committed:
python3Packages.taskw: 1.2.0 -> 1.3.0

FRidh committed:
python3Packages.tablib: 2.0.0 -> 3.0.0

FRidh committed:
python3Packages.thinc: 7.4.5 -> 8.0.2

FRidh committed:
python3Packages.tabulate: 0.8.8 -> 0.8.9

FRidh committed:
python3Packages.sysv_ipc: 1.0.1 -> 1.1.0

FRidh committed:
python3Packages.tenacity: 6.3.1 -> 7.0.0

FRidh committed:
python3Packages.titlecase: 1.1.1 -> 2.0.0

FRidh committed:
python3Packages.terminado: 0.9.2 -> 0.9.3

FRidh committed:
python3Packages.supervisor: 4.2.1 -> 4.2.2

FRidh committed:
python3Packages.tifffile: 2020.11.26 -> 2021.3.17

FRidh committed:
python3Packages.tweepy: 3.9.0 -> 3.10.0

FRidh committed:
python3Packages.typeguard: 2.10.0 -> 2.11.1

FRidh committed:
python3Packages.trimesh: 3.9.7 -> 3.9.10

FRidh committed:
python3Packages.treq: 20.9.0 -> 21.1.0

FRidh committed:
python3Packages.trustme: 0.6.0 -> 0.7.0

FRidh committed:
python3Packages.trytond: 5.8.4 -> 5.8.5

FRidh committed:
python3Packages.tqdm: 4.58.0 -> 4.59.0

FRidh committed:
python3Packages.transmission-rpc: 3.2.2 -> 3.2.4

FRidh committed:
python3Packages.traits: 6.1.1 -> 6.2.0

FRidh committed:
python3Packages.twill: 2.0.2 -> 3.0

FRidh committed:
python3Packages.txaio: 20.4.1 -> 21.2.1

FRidh committed:
python3Packages.tox: 3.20.1 -> 3.23.0

FRidh committed:
python3Packages.Unidecode: 1.1.2 -> 1.2.0

FRidh committed:
python3Packages.twine: 3.2.0 -> 3.4.1

FRidh committed:
python3Packages.uproot3-methods: 0.10.0 -> 0.10.1

FRidh committed:
python3Packages.xlrd: 1.2.0 -> 2.0.1

FRidh committed:
python3Packages.waitress: 1.4.4 -> 2.0.0

FRidh committed:
python3Packages.xml2rfc: 3.5.0 -> 3.6.0

FRidh committed:
python3Packages.xarray: 0.16.2 -> 0.17.0

FRidh committed:
python3Packages.PyVirtualDisplay: 2.0 -> 2.1

FRidh committed:
python3Packages.west: 0.8.0 -> 0.10.1

FRidh committed:
python3Packages.wcmatch: 8.1.1 -> 8.1.2

FRidh committed:
python3Packages.yamlloader: 0.5.5 -> 1.0.0

FRidh committed:
python3Packages.uritools: 3.0.0 -> 3.0.1

FRidh committed:
python3Packages.virtualenv: 20.2.2 -> 20.4.3

FRidh committed:
python3Packages.zope.testing: 4.7 -> 4.9

FRidh committed:
python3Packages.zope.schema: 6.0.1 -> 6.1.0

FRidh committed:
python3Packages.zope.interface: 5.2.0 -> 5.3.0

FRidh committed:
python3Packages.zope.component: 4.6.2 -> 5.0.0

FRidh committed:
python3Packages.zstandard: 0.15.1 -> 0.15.2

FRidh committed:
python3Packages.zope.testrunner: 5.2 -> 5.3.0

FRidh committed:
python3Packages.zipp: 3.4.0 -> 3.4.1

FRidh committed:
python3Packages.zopfli: 0.1.7 -> 0.1.8

FRidh committed:
python2Packages.setuptools-scm: keep 5.0.2

FRidh committed:
python2Packages.idna: keep 2.10

FRidh committed:
python2Packages.chardet: keep 3.0.4

FRidh committed:
python2Packages: separate extension

FRidh committed:
python2Packages.pytestrunner: keep 5.2

FRidh committed:
python2Packages: move python 2 calls into python2-packages.nix

SuperSandro2000 commented:
At least https://github.com/NixOS/nixpkgs/pull/117401 and probably some more sage related dependencies shouldn't be updated.

FRidh committed:
python3Packages.greenlet: disable tests

collares commented:
@SuperSandro2000 Thanks for the heads up. If this gets merged anyway, I will just rebase my PR to remove the `cypari2` update and just fix Sage test expectations.

FRidh committed:
python3Packages.mpmath: add missing dep

FRidh committed:
python3Packages.xlrd: disable tests

FRidh committed:
python3Packages.lazy-object-proxy: remove upper constraint

FRidh committed:
pythonPackages: inherit stdenv as well

FRidh committed:
python2Packages.arrow: keep 0.17.0

FRidh committed:
python3Packages.gitdb: allow never version smmap

Let's hope nothing breaks

FRidh committed:
pythonPackages.singledispatch: fix build

FRidh committed:
python3Packages.fasteners: fix build

FRidh committed:
python3Packages.importlib-metadata: 3.7.2 -> 3.7.3

FRidh committed:
python2Packages.pyroma: keep 2.6.1

FRidh committed:
sage: keep cypari2 at 2.1.1

FRidh committed:
python3Packages.requests: handle idna 3

FRidh committed:
python3Packages.flit-core: 2.3.0 -> 3.2.0

FRidh committed:
python3Packages.flit: 3.0.0 -> 3.2.0

mweinelt committed:
python3Packages.idna: enable tests

mweinelt committed:
python27Packages.idna: enable tests

FRidh committed:
pythonPackages: move python 2 only expressions to python2-packages.nix

FRidh commented:
/rebase staging

jonringer commented:
ah, I didn't miss this one!

FRidh commented:
Python 2 packages now live in `python2-packages.nix`. The expressions are however still in `python-modules`. Some day those could be moved to `python2-modules`....or just be deleted.

jonringer committed:
python3{6,7}.importlib-metadata: fix build

FRidh commented:
PyPI updates are included here, but we now have a significant portion of packages that we fetch from GitHub. Should run `nix-update` over it once as well.

jonringer commented:
> PyPI updates are included here, but we now have a significant portion of packages that we fetch from GitHub. Should run `nix-update` over it once as well.

The default update script for buildPythonPackage is able to query github as well.

jonringer commented:
rebased on latest staging and resolved conflicts. Had to force push though, so please keep that in mind

jonringer committed:
python3Packages.pyflakes: 2.3.0 -> 2.3.1

jonringer committed:
python3Packages.yfinance: 0.1.55 -> 0.1.56

jonringer committed:
python3Packages.aws-xray-sdk: 2.6.0 -> 2.7.0

jonringer committed:
python3Packages.pyperf: 2.1.0 -> 2.2.0

jonringer committed:
python3Packages.pysma: 0.4 -> 0.4.1

jonringer committed:
python3Packages.binwalk: 2.2.0 -> 27

jonringer committed:
python3Packages.brotli: 1.0.7 -> 1.0.9

jonringer committed:
python3Packages.aws-lambda-builders: 1.1.0 -> 1.3.0

jonringer committed:
python3Packages.cocotb: 1.4.0 -> 1.5.1

jonringer committed:
python3Packages.construct: 2.10.56 -> 2.10.63

jonringer committed:
python3Packages.cfn-lint: 0.42.0 -> 0.48.0

jonringer committed:
python3Packages.chart-studio: 1.1.0 -> 4.14.3

jonringer committed:
python3Packages.exchangelib: 3.2.1 -> 4.1.0

jonringer committed:
python3Packages.flask-mongoengine: 0.9.5 -> 1.0.0

jonringer committed:
python3Packages.gssapi: 1.6.10 -> 1.6.12

jonringer committed:
python3Packages.gsd: 1.9.3 -> 2.4.1

jonringer committed:
python3Packages.habanero: 0.6.0 -> 0.7.4

jonringer committed:
python3Packages.javaproperties: 0.7.0 -> 0.8.0

jonringer committed:
python3Packages.hyppo: 0.1.3 -> 0.2.1

jonringer committed:
python3Packages.ignite: 0.4.2 -> 0.4.4.post1

jonringer committed:
python3Packages.libgpuarray: 0.7.5 -> 0.7.6

jonringer committed:
python3Packages.keepkey: 6.6.0 -> 6.7.0

jonringer committed:
python3Packages.ipfsapi: 0.4.2.post1 -> 0.7.0

jonringer committed:
python3Packages.markdown-it-py: 0.5.6 -> 0.6.2

jonringer committed:
python3Packages.mongoengine: 0.22.0 -> 0.23.0

jonringer committed:
python3Packages.liquidctl: 1.4.2 -> 1.5.1

jonringer committed:
python3Packages.monty: 3.0.2 -> 2021.3.3

jonringer committed:
python3Packages.optuna: 0.19.0 -> 2.6.0

jonringer committed:
python3Packages.mail-parser: 3.12.0 -> 3.15.0

jonringer committed:
python3Packages.mne-python: 0.21.2 -> 0.22.0

jonringer committed:
python3Packages.pc-ble-driver-py: 0.14.2 -> 0.15.0

jonringer committed:
python3Packages.pre-commit-hooks: 3.3.0 -> 3.4.0

jonringer committed:
python3Packages.pubnub: 4.8.0 -> 5.1.0

jonringer committed:
python3Packages.perfplot: 0.8.4 -> 0.8.10

jonringer committed:
python3Packages.pybind11: 2.6.1 -> 2.6.2

jonringer committed:
python3Packages.pycairo: 1.18.2 -> 1.20.0

jonringer committed:
python3Packages.pydub: 0.24.1 -> 0.25.1

jonringer committed:
python3Packages.pygccxml: 1.9.1 -> 2.1.0

jonringer committed:
python3Packages.pyflume: 0.6.2 -> 0.6.4

jonringer committed:
python3Packages.pycm: 2.5 -> 3.1

jonringer committed:
python3Packages.pydantic: 1.8 -> 1.8.1

jonringer committed:
python3Packages.pymodbus: 2.4.0 -> 2.5.0

jonringer committed:
python3Packages.pyparted: 3.11.4 -> 3.11.7

jonringer committed:
python3Packages.pynetdicom: 1.5.5 -> 1.5.6

jonringer committed:
python3Packages.pylatexenc: 2.8 -> 2.9

jonringer committed:
python3Packages.pygmt: 0.2.0 -> 0.3.1

jonringer committed:
python3Packages.pylint-django: 2.3.0 -> 2.4.2

jonringer committed:
python3Packages.python-frontmatter: 0.5.0 -> 1.0.0

jonringer committed:
python3Packages.robotframework-seleniumlibrary: 3.3.1 -> 5.1.3

jonringer committed:
python3Packages.qds_sdk: 1.15.2 -> 1.16.1

jonringer committed:
python3Packages.pytest-ansible: 2.1.1 -> 2.2.3

jonringer committed:
python3Packages.rq: 1.5.2 -> 1.7.0

jonringer committed:
python3Packages.rich: 9.10.0 -> 9.13.0

jonringer committed:
python3Packages.python-snap7: 0.11 -> 1.0

jonringer committed:
python3Packages.pytorch-metric-learning: 0.9.95 -> 0.9.97

jonringer committed:
python3Packages.python-lz4: 2.1.10 -> 3.1.3

jonringer committed:
python3Packages.pytest-helpers-namespace: 2019.1.8 -> 2021.3.24

jonringer committed:
python3Packages.sumo: 1.0.9 -> 2.2.1

jonringer committed:
python3Packages.symengine: 0.6.1 -> 0.7.0.post3

jonringer committed:
python3Packages.simple-salesforce: 0.74.3 -> 1.11.0

jonringer committed:
python3Packages.shap: 0.36.0 -> 0.39.0

jonringer committed:
python3Packages.seabreeze: 0.6.0 -> 1.3.0

jonringer committed:
python3Packages.stumpy: 1.4 -> 1.8.0

jonringer committed:
python3Packages.signedjson: 1.1.0 -> 1.1.1

jonringer committed:
python3Packages.sshpubkeys: 3.1.0 -> 3.3.1

jonringer committed:
python3Packages.torchvision: 0.8.2 -> 0.9.0

jonringer committed:
python3Packages.xmlschema: 1.4.1 -> 1.5.3

jonringer committed:
python3Packages.tinydb: 4.1.1 -> 4.4.0

jonringer committed:
python3Packages.tensorboardx: 2.0 -> 2.1

jonringer commented:
did an update on github packages as well, I dropped the `hypothesis` and `pyproj` bumps because I feel like that would cause too much effort to stabilize

FRidh commented:
What do you think about upgrading sphinx? Might cause some breakage

jonringer committed:
python3Packages.javaproperties: fix build

jonringer commented:
> What do you think about upgrading sphinx? Might cause some breakage

Probably be okay, we can use staging-next to find other regressions

jonringer committed:
python3Packages.brotli: fix tests

jonringer committed:
python3Packages.Fabric: fix build

dotlambda commented:
> Python 2 packages now live in `python2-packages.nix`.

You forgot `wsproto`.

jonringer committed:
python3Packages.construct: fix build and tests

jonringer committed:
python3Packages.jupyter-packaging: fix src and build

jonringer commented:
was trying to fix the jupyter packages, turned into a major PITA, current progress is captured: https://github.com/NixOS/nixpkgs/pull/117506

May just revert jupyter bumps for this update set, unless someone else wants to try their hand.

jonringer committed:
python3Packages.keyring: fix build

jonringer committed:
azure-cli: fix build

jonringer committed:
python3Packages.poetry: 1.1.14 -> 1.1.15, fix build

dotlambda commented:
Let's move the `python-igraph` bump to https://github.com/NixOS/nixpkgs/pull/117512.

jonringer committed:
python3Packages.pytest: fix tests

collares commented:
`cypari2` was just updated in master.

FRidh committed:
python3Packages.jaraco_itertools: remove patch

No longer applies and we don't test anymore anyway.

FRidh committed:
python2Packages.typing: we no longer need it for Python 3

FRidh committed:
python3Packages.ipfsapi: fix eval

FRidh committed:
python2Packages.scandir: python 2 only

FRidh committed:
python2Packages.singledispatch: python 2 only

FRidh committed:
python3Packages.bleachbit: fix eval

FRidh commented:
There are quite some evaluation errors because Python 2 was no longer evaluated...

FRidh committed:
python2Packages.pytest-xdist: fix attribute name and alias

FRidh committed:
python2Packages.convertdate: fix attribute name

FRidh committed:
gcalcli: fix eval

FRidh committed:
buildPythonPackage: avoid recursion

Sometimes an infinite recursion error shows up.

mweinelt committed:
python3Packages.pywebpush: fix build

Updates dependencies and migrates test from nosetests to pytest.

mweinelt committed:
python3Packages.josepy: fix test run and drop unused dependency

mweinelt committed:
python3Packages.pymazda: fix build

The pycryptodome dependency was replaced by cryptography. Sadly the git
repository is out of date.

mweinelt committed:
python3Packages.python-smarttub: pin pyjwt at 1.7.1

Upstream does not yet support pyjwt>=2.0 and this package is only used
in the home-assistant context, which also pins 1.7.1.

https://github.com/mdz/python-smarttub/issues/10

mweinelt committed:
home-assistant: pin pyjwt==1.7.1, pykmtronic==0.0.3

mweinelt commented:
Fixed up the home-assistant build.

FRidh committed:
python3Packages.eth-utils: 1.9.5 -> 1.10.0

FRidh committed:
python3Packages.eth-hash: break circular dependency

FRidh committed:
calibre-web: don't require singledispatch

which is a Python 2 only package.

FRidh committed:
quodlibet: remove pyobjc as dependency

because its python 2 only and quodlibet uses python3

FRidh committed:
Merge branch 'staging' into python-unstable

FRidh commented on pkgs/development/python-modules/python-smarttub/default.nix:
```diff
@@ -26,7 +37,7 @@ buildPythonPackage rec {
     aiohttp
     inflection
-    pyjwt
+    pyjwt'
```
Don't ever do this. What happens now is that within a closure, you can have two versions of pyjwt. This should absolutely not happen.

The solution is to override the python3 package set that home-assistant uses.

FRidh replied:
Reverted in d9fd85f9fb883f75e2000785116f2480a40a687f and marked the package as conditionally broken. Annoying for users of the package that don't use it through ha, but there is no way around it.

FRidh commented:
Further fixing will be done in staging-next. Pandas is currently broken there which makes it anyway difficult to see what needs further care.

SuperSandro2000 commented:
Maybe this helps https://github.com/NixOS/nixpkgs/pull/118029/files. I had some hanging tests but maybe targeted the wrong branch.

dotlambda commented:
@jonringer Your pycairo bump broke `python2.pkgs.pycairo`: https://github.com/NixOS/nixpkgs/issues/119174
We'll need to package the last version with support for Python 2.
